weave/trace_server/image_completion.py

```
@@ -0,0 +1,215 @@
+            )
+    """
+    from litellm import image_generation
+
+    model_name = inputs.get("model", "")
+
+    if not model_name:
+        raise InvalidRequest("Model name is required")
+
+    try:
+        # Make the image generation API call
+        # Pass all parameters so LiteLLM can create proper model identifier for cost calculation
+        image_params = {
+            "model": inputs.get("model"),
+            "prompt": inputs.get("prompt"),
+            "api_key": api_key,
+        }
+
+        # Add image generation parameters based on model capabilities
+
+        # Handle 'n' parameter (number of images)
+        # DALL-E 3 only supports n=1, others can support multiple
+        if model_name == "dall-e-3":
+            image_params["n"] = 1  # DALL-E 3 only supports 1 image
+        else:
+            image_params["n"] = inputs.get("n", 1)
+
+        # Only set quality parameter for models that support it (DALL-E 3 and gpt-image-1)
+        if model_name in ["dall-e-3", "gpt-image-1"]:
+            default_quality = "high" if model_name == "gpt-image-1" else "standard"
+            image_params["quality"] = default_quality
+
+        # Only set style parameter for DALL-E 3
+        if model_name == "dall-e-3":
+            image_params["style"] = "natural"  # Default style
+
+        # Set default size for models that need it
+        if model_name == "dall-e-3":
+            image_params["size"] = "1024x1024"
+        elif model_name == "gpt-image-1":
+            image_params["size"] = "1024x1024"
+        else:
+            image_params["size"] = "1024x1024"
+
+        try:
+            # gpt-image-1 doesn't support response_format parameter
+            if model_name == "gpt-image-1":
+                res = image_generation(**image_params)
+            else:
+                res = image_generation(response_format="url", **image_params)
+        except Exception as e:
+            raise
+
```

I think these parameters should be sent in the request and validated by the client. We should throw an error if the params are not valid (can probably just propagate whatever the API sends back to the user